packages/costa/src/proto.ts

```diff
@@ -0,0 +1,23 @@
+export interface PluginMessage {
+  event: string;
+  params?: any[];
+}
+
+export enum PluginOperator {
+  START = 0x30,
+  WRITE = 0x31,
+  READ = 0x32,
+  COMPILE = 0x100
+}
+
+export class PluginProto {
+  op: PluginOperator;
```

Comment:
Maybe 'operator' is better just like 'message' below?

Reply:
The message is called PluginMessage, either.